db/src/test/java/com/yahoo/maha/data/StringEventBatchDecoder.java

```diff
@@ -0,0 +1,96 @@
+// Copyright 2017, Yahoo Holdings Inc.
+// Licensed under the terms of the Apache License 2.0. Please see LICENSE file in project root for terms.
+package com.yahoo.maha.data;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.lang3.concurrent.ConcurrentException;
+import org.apache.commons.lang3.concurrent.LazyInitializer;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * Created by hiral on 8/2/18.
+ */
+public class StringEventBatchDecoder implements Decoder<StringEventBatch> {
+
+    private final int bufferSize;
+    private final LazyInitializer<ByteBuffer> lazyInputByteBuffer;
+    private final LazyInitializer<ByteBuffer> lazyOutputByteBuffer;
+    private final LazyInitializer<byte[]> lazyOutputByteArray;
+    private final Compressor compressor;
+
+    StringEventBatchDecoder(Compressor.Codec codec) {
+        this(codec, 1);
+    }
+    StringEventBatchDecoder(Compressor.Codec codec, int bufferMb) {
+        bufferSize = bufferMb * 1024 * 1024;
```

Comment:
FileUtils has some readable constants for MB, GB etc https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html